eng/Versions.props

```diff
@@ -72,9 +73,8 @@
     <Tooling_MicrosoftCodeAnalysisTestingVersion>1.1.2-beta1.22512.1</Tooling_MicrosoftCodeAnalysisTestingVersion>
     <MicrosoftVisualStudioShellPackagesVersion>17.5.0-preview-2-33117-317</MicrosoftVisualStudioShellPackagesVersion>
     <MicrosoftVisualStudioPackagesVersion>17.5.274-preview</MicrosoftVisualStudioPackagesVersion>
-    <RoslynPackageVersion>4.6.0-3.23164.6</RoslynPackageVersion>
+    <RoslynPackageVersion>$(MicrosoftNetCompilersToolsetVersion)</RoslynPackageVersion>
```

Comment:
Hold up a second. I'm very concerned by the direction this change is going: we're talking about two completely unrelated concepts here. The toolset compiler is the version of the compiler that is used to build this repository. `RoslynPackageVersion` determines what Roslyn APIs the razor compiler and tooling expect when being run. Dynamically moving this forward for source-build would mean that we're compiling a configuration that we don't test.

Reply:
> We can't actually use proxy properties like this. Product source build won't be able to override `RoslynPackageVersion` since no package has that name. So, in source-build, any dependency using `RoslynPackageVersion` will retain its original value, leading to prebuilts.

How does this work today, is this already an issue? Bunch of properties are initialized with value of `RoslynPackageVersion` - https://github.com/dotnet/razor/blob/625b592c6f2adc55c4af94fda0062afcfbb1df14/eng/Versions.props#L138-L151

Reply:
My concern is that the interaction between that version of roslyn's APIs and Razor has not been tested. I don't care so much about what version of the toolset compiler is used to compile Razor; I care much more deeply about what version of Roslyn we're referencing when we build.

Reply:
Source-build doesn't really have an option there. It has to build razor based on the version of roslyn that ships in the SDK.

Reply:
> How does this work today, is this already an issue? Bunch of properties are initialized with value of `RoslynPackageVersion` -

I'm not entirely sure...I would expect those to show up as prebuilts. @dotnet/source-build-internal?

Reply:
> Source-build doesn't really have an option to bifurcate these.

OK - so the suggestion is to replace all instances of $(RoslynPackageVersion) with $(MicrosoftNetCompilersToolsetVersion) - I'll do that.

Reply:
Looking through a few of them, I think the properties are all used for PackageReferences that are excluded from the Linux source build.

Reply:
So, in that case @NikolaMilosavljevic, we should leave the RoslynPackageVersion properties alone and use add a property/dependency for Microsoft.Net.Compilers.Toolset, like you did before.

If these projects ever get pulled into Linux source build, this will have to be resolved.

Reply:
So essentially, source build is **only** overriding M.N.C.T version.

Reply:
`Microsoft.CodeAnalysis.Common` and `Microsoft.CodeAnalysis.CSharp` were showing as prebuilts until I synchronized `$(RoslynPackageVersion)` with `$(MicrosoftNetCompilersToolsetVersion)`. As mentioned before, source-build needs to consume packages from a single `roslyn` intermediate package, so versions naturally have to be the same.

Reply:
Can you root cause why that synchronization is necessary (where the package refs are used). They do have to be synchronized if the package refs show up in source build. But it looks like most of refs were in projects excluded from source build, which would not affect the pre-builts.